docs-next/pages/apis/context.mdx

```diff
@@ -2,6 +2,139 @@ import { Markdown } from '../../components/Page';
 
 # Context API
 
-(coming soon)
+The `KeystoneContext` type represents an [Apollo Server context](https://www.apollographql.com/docs/apollo-server/data/resolvers/#the-context-argument) object.
+A `KeystoneContext` object is made available to many of the server-side functions in a Keystone system.
+It contains the APIs required to write the business logic for things like [access control](../guides/access-control), [hooks](../guides/hooks), [testing](../guides/testing) and GraphQL [schema extensions](../guides/schema-extension).
+
+```typescript
+import type { KeystoneContext } from '@keystone-next/types';
+
+context = {
+  // HTTP request object
+  req,
+
+  // List item API
+  lists,
+
+  // GraphQL helpers
+  graphql: {
+    schema,
+    run,
+    raw,
+  };
+
+  // Session API
+  session,
+  startSession,
+  endSession,
+
+  // New context creators
+  sudo,
+  exitSudo,
+  withSession,
+
+  // Database access
+  knex,
+  prisma,
+  mongoose,
+
+  // Access control helpers
+  getListAccessControlForUser,
+  getFieldAccessControlForUser,
+
+  // Internal state
+  totalResults,
+  maxTotalResults,
+  schemaName,
+
+  // Deprecated
+  gqlNames,
+  executeGraphQL,
+  keystone,
+};
+```
+
+### HTTP request object
+
+`req`: The [IncomingMessage](https://nodejs.org/api/http.html#http_class_http_incomingmessage) object from the HTTP request which called the GraphQL API.
+
+### List item API
+
+`lists`: The [list items API](./list-items), which can be used to perform CRUD operations against your GraphQL API.
+
+### GraphQL helpers
+
+`graphql.schema`: The [GraphQL Schema](https://graphql.org/graphql-js/type/#graphqlschema) object, which can be used directly, or via `graphql.raw`, `graphql.run`, or the `lists` API.
```

Comment:
I don't know if it's too much here (and maybe we can come back to this later)

But I feel some short-and-sweet examples here would make this much more understandable.

Like

```ts
(context) => {
  const result = graphql.run({ query, variables });
}
```

It's just kind of hard to see how it all fits together or how you'd use it like this.

Reply:
This isn't really the right place for examples, but we should definitely link out to the guides where we have room to talk people through how to use these things for different stuff 👍